test/e2e/gang_scheduling/gang_scheduling_test.go

```diff
@@ -899,119 +498,231 @@ var _ = Describe("", func() {
+			TaskGroups: []interfaces.TaskGroup{
+				{Name: "GROUPCAPS", MinMember: int32(3), MinResource: minResource},
 			},
-			Resources: &v1.ResourceRequirements{
-				Requests: v1.ResourceList{
-					"cpu":    podResources["cpu"],
-					"memory": podResources["memory"],
-				},
+		}),
+		ginkgo.Entry("Verify_TG_With_Invalid_MinMember", k8s.PodAnnotation{
+			TaskGroups: []interfaces.TaskGroup{
+				{Name: groupA, MinMember: int32(-1), MinResource: minResource},
 			},
-		}
+		}),
+	)
 
-		podTest, err := k8s.InitTestPod(podConf)
-		Ω(err).NotTo(HaveOccurred())
-		taskGroupsMap, annErr := k8s.PodAnnotationToMap(podConf.Annotations)
-		Ω(annErr).NotTo(HaveOccurred())
-		By(fmt.Sprintf("Deploy pod %s with task-groups: %+v", podTest.Name, taskGroupsMap[k8s.TaskGroups]))
-		originator, err := kClient.CreatePod(podTest, ns)
+	// Test placeholder with hugepages
+	// 1. Deploy 1 job with hugepages-2Mi
+	// 2. Verify all pods running
+	It("Verify_HugePage", func() {
+		hugepageKey := fmt.Sprintf("%s2Mi", v1.ResourceHugePagesPrefix)
+		nodes, err := kClient.GetNodes()
 		Ω(err).NotTo(HaveOccurred())
+		hasHugePages := false
+		for _, node := range nodes.Items {
+			if v, ok := node.Status.Capacity[v1.ResourceName(hugepageKey)]; ok {
+				if v.Value() != 0 {
+					hasHugePages = true
+					break
+				}
+			}
+		}
+		if !hasHugePages {
+			ginkgo.Skip("Skip hugepages test as no node has hugepages")
+		}
 
-		By("Verify appState = Accepted")
-		timeoutErr := restClient.WaitForAppStateTransition(defaultPartition, nsQueue, podConf.Labels["applicationId"], yunikorn.States().Application.Accepted, 20)
-		Ω(timeoutErr).NotTo(HaveOccurred())
-
-		By("Wait for placeholders running")
-		phNames := yunikorn.GetPlaceholderNames(podConf.Annotations, podConf.Labels["applicationId"])
-		tgBNames := phNames[podConf.Annotations.TaskGroups[1].Name]
-		for _, ph := range tgBNames {
-			runErr := kClient.WaitForPodRunning(ns, ph, 30*time.Second)
-			Ω(runErr).NotTo(HaveOccurred())
+		// add hugepages to request
+		minResource[hugepageKey] = resource.MustParse("100Mi")
+		annotations := k8s.PodAnnotation{
+			TaskGroupName: groupA,
+			TaskGroups: []interfaces.TaskGroup{
+				{Name: groupA, MinMember: int32(3), MinResource: minResource},
+			},
 		}
+		job := createJob(appID, minResource, annotations, 3)
 
-		By("Delete originator pod")
-		deleteErr := kClient.DeletePod(originator.Name, ns)
-		Ω(deleteErr).NotTo(HaveOccurred())
+		By("Verify all job pods are running")
+		jobRunErr := kClient.WaitForJobPods(ns, job.Name, int(*job.Spec.Parallelism), 2*time.Minute)
+		Ω(jobRunErr).NotTo(HaveOccurred())
 
-		By("Verify placeholders deleted")
-		for _, placeholders := range phNames {
-			for _, ph := range placeholders {
-				deleteErr = kClient.WaitForPodTerminated(ns, ph, 30*time.Second)
-				Ω(deleteErr).NotTo(HaveOccurred(), "Placeholder %s still running", ph)
-			}
+		checkAppStatus(appID, yunikorn.States().Application.Running)
```

Comment:
Can we also show the resources used by pods and assert/check for expected resource type (huge pages in this case)? so that it will come out very clearly.

Reply:
Add following assertion to validate it. Thank you.
```
Ω(appDaoInfo.UsedResource[hugepageKey]).To(Equal(int64(314572800)), "Used huge page resource is not correct")
```